pywbem_mock/_defaultinstanceprovider.py

```diff
@@ -0,0 +1,439 @@
+            instance_store.create(new_instance.path, new_instance)
+        except ValueError:
+            raise CIMError(
+                CIM_ERR_ALREADY_EXISTS,
+                _format("NewInstance {0!A} already exists in namespace "
+                        "{1!A}.", new_instance.path, namespace))
+
+        # Create instance returns model path
+        return new_instance.path.copy()
+
+    ####################################################################
+    #
+    #   DeleteInstance server request
+    #
+    ####################################################################
+
+    def DeleteInstance(self, InstanceName):
+        """
+        Implements a WBEM server responder for
+        :meth:`pywbem.WBEMConnection.DeleteInstance`.
+
+        This method deletes a single instance from the CIM repository based on
+        the  InstanceName parameter.
+
+        It does not attempt to delete referencing instances (associations,
+        etc.) that reference this instance.
+
+        If the creation class of the instance to be deleted is PG_Namespace or
+        CIM_Namespace, then the namespace identified by the 'Name' property
+        of the instance is being deleted in the CIM repository in addition to
+        the CIM instance.
+
+        NOTE: This method includes namespace within the InstanceName
+        rather than as a separate input argument
+
+        Parameters:
+
+          InstanceName (:class:`~pywbem.CIMInstanceName`):
+            The instance path of the instance to be deleted.
+            The instance path of the instance to be retrieved  with the
+            following attributes:
+
+            * `classname`: Name of the creation class of the instance.
+            * `keybindings`: Keybindings of the instance.
+            * `namespace`: Name of the CIM namespace containing the instance.
+              Must not be None.
+            * `host`: value ignored.
+
+        Raises:
+            CIMError: CIM_ERR_INVALID_NAMESPACE
+            CIMError: CIM_ERR_INVALID_CLASS
+            CIMError: CIM_ERR_NOT_FOUND
+        """
+
+        instance_store = self.get_instance_store(InstanceName.namespace)
+
+        # Handle namespace deletion, currently hard coded.
+        # Issue #2062 TODO/AM 8/18 Generalize the hard coded handling into
+        # provider concept
+        classname_lower = InstanceName.classname.lower()
+        if classname_lower == 'pg_namespace':
+            ns_classname = 'PG_Namespace'
+        elif classname_lower == 'cim_namespace':
+            ns_classname = 'CIM_Namespace'
+        else:
+            ns_classname = None
+        if ns_classname:
+            namespace = InstanceName.keybindings['Name']
+
+            # Reflect the namespace deletion in the CIM repository
+            # This will check the namespace for being empty.
+            self.remove_namespace(namespace)
+
+        # Delete the instance from the CIM repository
+        instance_store.delete(InstanceName)
```

Comment:
This provider should also handle ModifyInstance.

Reply:
DONE